richtextfx/src/main/java/org/fxmisc/richtext/model/UndoActions.java

```diff
@@ -21,8 +20,8 @@
     default void redo() { getUndoManager().redo(); }
 
     default boolean isUndoAvailable() { return getUndoManager().isUndoAvailable(); }
-    default ObservableBooleanValue undoAvailableProperty() { return getUndoManager().undoAvailableProperty(); }
+    default Val<Boolean> undoAvailableProperty() { return getUndoManager().undoAvailableProperty(); }
 
     default boolean isRedoAvailable() { return getUndoManager().isRedoAvailable(); }
-    default ObservableBooleanValue redoAvailableProperty() { return getUndoManager().redoAvailableProperty(); }
+    default Val<Boolean> redoAvailableProperty() { return getUndoManager().redoAvailableProperty(); }
```

Comment:
Just FYI. I get a compiler warning on this and the corresponding undo line: "Unchecked assignment `Val` to `Val<Boolean>`. Reason `getUndoManager()` has raw type so result of `redoAvailableProperty()` is erased"

If I add the type parameter (`UndoActions<C>` and `UndoManager<C> getUndoManager()`), they go away. However, then I can't specify the type in `GenericStyledArea` since the option to create both a plain and rich undo manager exists.